packages/auth-sdk/user.tsx

```diff
@@ -16,16 +17,46 @@ const userKeys = {
    user: ['user'],
 };
 
+const hasLocalStorage = typeof localStorage !== 'undefined';
+const userDataLocalKey = 'user.data';
+
 export function useAuthenticatedUser() {
    const apiClient = useIFixitApiClient();
+   const cachedUserData = useMemo(
+      () =>
+         !hasLocalStorage
+            ? null
+            : (() => {
+                 const storedString = localStorage.getItem(userDataLocalKey);
+                 return storedString === null ? null : JSON.parse(storedString);
+              })(),
+      []
+   );
    const query = useQuery(
       userKeys.user,
-      () => fetchAuthenticatedUser(apiClient).catch(() => null),
+      () => {
+         const responsePromise = fetchAuthenticatedUser(apiClient).catch(
+            () => null
+         );
+         if (hasLocalStorage) {
+            responsePromise
+               .then((response) => {
+                  localStorage.setItem(
+                     userDataLocalKey,
+                     JSON.stringify(response)
+                  );
+               })
```

Comment:
I'm not super familiar with our user api. If `response` is ever undefined or an empty string or something, I think the `JSON.parse` above will throw. But unless we see problems in practice it probably doesn't matter

Reply:
Looks like just undefined is a problem:
![image](https://user-images.githubusercontent.com/26855/198355737-8f1ddfa1-0d2e-49d2-a351-f6b0e2a97532.png)

I don't think the above function can ever return *undefined* (because typescript), and if it does, we have the `catch()` below to turn that into null.